consensus-types/blocks/getters.go

```diff
@@ -29,11 +29,59 @@ func (b *SignedBeaconBlock) Signature() [field_params.BLSSignatureLength]byte {
 	return b.signature
 }
 
+// SetSignature sets the signature of the signed beacon block.
+func (b *SignedBeaconBlock) SetSignature(sig []byte) {
+	copy(b.signature[:], sig)
+}
+
 // Block returns the underlying beacon block object.
 func (b *SignedBeaconBlock) Block() interfaces.BeaconBlock {
 	return b.block
 }
 
+// SetBlock sets the underlying beacon block object.
+func (b *SignedBeaconBlock) SetBlock(blk interfaces.BeaconBlock) error {
```

Comment:
We want a copy here, right? You can create a `Copy()` method on `type BeaconBlock interface` and call it at the beginning here.